docs/spec.md

```diff
@@ -0,0 +1,351 @@
+   # + port - port of the Redis database
+   # + username - field description
+   # + password - The password for the Redis database
+   # + options - other connection options of the connection configuration
+   type ConnectionParams record {|
+     string host = "localhost";
+     int port = 6379;
+     string username?;
+     string password?;
+     Options options = {};
+   |};
+   ```
+
+  ```ballerina
+  # Connection options for Redis client endpoint.
+  #
+  # + secureSocket - configurations related to SSL/TLS encryption
+  # + clientName - name of the client
+  # + database - database index
+  # + connectionTimeout - connection timeout in seconds
+  public type Options record {|
+      SecureSocket secureSocket?;
+      string clientName?;
+      int database = -1;
+      int connectionTimeout = 60;
+  |};
+  ```
+  ```ballerina  
+  # Configurations for secure communication with the Redis server.
+  #
+  # + cert - configurations associated with `crypto:TrustStore` or single certificate file that the client trusts
+  # + key - configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client
+  # + protocols - list of protocols used for the connection established to Redis Server, such as TLSv1.2, TLSv1.1, TLSv1.
+  # + ciphers - list of ciphers to be used for SSL connections
+  # + verifyPeer - whether peer verification is enabled
+  # + startTls - whether StartTLS is enabled
+  public type SecureSocket record {|
+      crypto:TrustStore|string cert;
+      crypto:KeyStore|CertKey key?;
+      string[] protocols?;
+      string[] ciphers?;
+      boolean verifyPeer?;
+      boolean startTls?;
+  |};
+  ```
+  ```ballerina
+  # Represents a combination of certificate, private key, and private key password if encrypted.
+  #
+  # + certFile - file containing the certificate
+  # + keyFile - file containing the private key in PKCS8 format
+  # + keyPassword - Password of the private key if it is encrypted
+  @display {label: "Certificate Key Configurations"}
+  public type CertKey record {|
+      string certFile;
+      string keyFile;
+      string keyPassword?;
+  |};
+  ``` 
+
+## 2.2. Handle connection pools
+
+The existing connection pooling implementation for Redis is based on the [Apache Commons Pool](https://commons.apache.org/proper/commons-pool/)
+library.
+
+The `ConnectionConfig` has a `connectionPooling` field that can be set to `true` to enable connection pooling.
+Currently, the connection pooling has limited configuration options, and the default pool size and other configurations
+of the underlying connection pool are used by the client.
+
+```ballerina
+
+## 2.3. Close the client
+
+The `close()` operation shuts down the Redis client and closes the associated connection pool.
+
+```ballerina
+# Closes the Redis client and shuts down the connection pool.
+#
+# + return - Possible error when closing the client
+public isolated function close() returns Error?;
+```
```

Comment:
I think we have an additional ```ballerina, shall we remove this?

Reply:
Removed with https://github.com/ballerina-platform/module-ballerinax-redis/pull/219/commits/5aec5d13ea49f791379343d8759d3c4819f987ad